pkg/controllers/nodepool/readiness/suite_test.go

```diff
@@ -88,12 +88,6 @@ var _ = Describe("Readiness", func() {
 		nodePool = ExpectExists(ctx, env.Client, nodePool)
 		Expect(nodePool.StatusConditions().Get(status.ConditionReady).IsFalse()).To(BeTrue())
 	})
-	It("should have status condition on nodePool as ready if nodeClass is ready", func() {
```

Comment:
Since operatorpkg has now changes for observed generation, nodePool status condition will not go ready unless the dependents are also set to true within the reconciliation lopp.

Reply:
IMO: We shouldn't drop this test just because of this. Can we set the dependents for our other sub-conditions to true so that this test passes?